hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmSnapshotManager.java

```diff
@@ -681,6 +686,14 @@ public SnapshotDiffResponse getSnapshotDiffReport(final String volume,
     return snapshotDiffReport;
   }
 
+  public List<SnapshotDiffJob> getSnapshotDiffList(final String volumeName,
+                                                   final String bucketName,
+                                                   final String jobStatus,
+                                                   final boolean listAll) {
+    return snapshotDiffManager.getSnapshotDiffJobList(
```

Comment:
I think we should validate that `volumeName` and `volumeName` exist to fail fast? May also add snapshot existed check.

Reply:
The only way I can think of to validate snapshots at this point, is to iterate the snapshot table. I think it's unnecessary since we are gonna get an empty list back, if there are no snapshots.

Reply:
I think you can use [path level SnapshotChain](https://github.com/apache/ozone/blob/master/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/SnapshotChainManager.java#L51).

Reply:
Done, I've also added an integration test to verify the error responses.